packages/angular-cli/upgrade/version.ts

```diff
@@ -105,11 +105,11 @@ export class Version {
       if (v.isLocal()) {
         console.warn(yellow('Using a local version of angular. Proceeding with care...'));
       } else {
-        if (v.major != 2 || ((v.minor == 3 && v.patch == 0) || v.minor < 3)) {
+        if (v.major < 2 || ((v.minor == 3 && v.patch == 0) || v.minor < 3)) {
```

Comment:
But there will be no Angular 3, right? But you brought up a good point because this will fail in ng 4.3.0. I'll fix it.